classes/Party.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * Party Class
+ *
+ * @package TheyWorkForYou
+ */
+
+namespace MySociety\TheyWorkForYou;
+
+/**
+ * Party
+ */
+
+class Party {
+
+    public $name;
+
+    private $db;
+
+    public function __construct($name) {
+        $this->name = $name;
+        $this->db = new \ParlDB;
+    }
+
+
+    public function getAllPolicyPositions($policies) {
+        $positions = $this->fetchPolicyPositionsByMethod($policies, "policy_position");
+
+        return $positions;
+    }
+
+    public function policy_position($policy_id, $want_score = false) {
+        $position = $this->db->query(
+            "SELECT score
+            FROM partypolicy
+            WHERE
+                party = :party
+                AND house = 1
+                AND policy_id = :policy_id",
+            array(
+                ':party' => $this->name,
+                ':policy_id' => $policy_id
+            )
+        );
+
+        if ( $position->rows ) {
+            $score = $position->field(0, 'score');
+            $score_desc = score_to_strongly($score);
+
+            if ( $want_score ) {
+                return array( $score_desc, $score);
+            } else {
+                return $score_desc;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    public function calculateAllPolicyPositions($policies) {
+        $positions = $this->fetchPolicyPositionsByMethod($policies, "calculate_policy_position");
+
+        return $positions;
+    }
+
+    public function calculate_policy_position($policy_id, $want_score = false) {
+
+        // This could be done as a join but it turns out to be
+        // much slower to do that
+        $divisions = $this->db->query(
+            "SELECT division_id, policy_vote, division_date
+            FROM policydivisions
+            WHERE policy_id = :policy_id
+            AND house = 'commons'",
+            array(
+                ':policy_id' => $policy_id
+            )
+        );
+
+        $score = 0;
+        $max_score = 0;
+        $scores = array();
+        $num_divs = $divisions->rows;
+        $total_votes = 0;
+
+        for ( $i = 0; $i < $num_divs; $i++ ) {
+            $division_id = $divisions->field($i, 'division_id');
+            $weights = $this->get_vote_scores($divisions->field($i, 'policy_vote'));
+
+            $votes = $this->db->query(
+                "SELECT count(*) as num_votes, vote
+                FROM persondivisionvotes
+                JOIN member ON persondivisionvotes.person_id = member.person_id
+                WHERE
+                    party = :party
+                    AND member.house = 1
+                    AND division_id = :division_id
+                    AND left_house = '9999-12-31'
```

Comment:
It does a little in that you're not comparing someone with how other members of their party voted at the time but I've taken the view that it's more useful to say how the MP compares to their current colleagues.

My hypothetical justification for this is if at an election the new intake causes a party to shift position on something and your MP doesn't agree with that shift then that's more useful to know than the fact that they voted in line with the previous position of the party.